### ISSUE TITLE
The Eye.json.tar.gz index archive corrupt or insufficient permissions

### ISSUE THREAD

**Issue by patredtheknight** [harmtemolder/calibre-store-the-eye#2]
Dear Harm, 
1. after sucessfully installing the latest plugin version I went to configure the plugin and clicked  the update the index button. 
2. this downloads an The Eye.json.tar.gz file to my ~/.config/calibre/config folder (size is always 2.5 MB) permissions: are -rw-rw-r-- for the correct user.
3. running the plugin in calibre gives no result for book or author
4. When I tried to untar the index file I got the following remark: 
```
$ tar -xvzf The\ Eye.json.tar.gz 
tar: This does not look like a tar archive
tar: Skipping to next header
tar: Exiting with failure status due to previous errors
```
Using a GUI (Ark) returns: Archive corrupted or insufficient permissions, even after changing the permissions

Is this normal behaviour? or is something wrong?

Thank you

**Comment by harmtemolder** [comment]
Oops, sounds like something’s wrong. I’ll dive into why this is happening, be right back

**Comment by harmtemolder** [comment]
@patredtheknight I believe I fixed your error. Could you test [v0.2.1-alpha](https://github.com/harmtemolder/calibre-store-the-eye/releases/tag/v0.2.1-alpha)?

**Comment by patredtheknight** [comment]
@harmtemolder 
1. I have downloaded the alpha release, unzipped, make (had to add releases folder) installed the plugin again and restarted.
2. I updated the index through the GUI and the file size is still 2.5 MB. I can now gunzip the file and get a 15 mb json (that I can't open in kate or vim because: 
```
The Eye.json was opened and contained lines longer than the configured Line Length Limit (10.000 characters).
The longest of those lines was 16.141.915 characters long
```
Those lines were wrapped and the document is set to read-only mode, as saving will modify its content.

3. HOWEVER: when I go to get books and search in the title field I do get some results and I am able to download the chosen book. Searching the author or keyword field gives no results. typing an author in the title field does get a result, probably because of the way The Eye works.

So except for the fact that I don't know if i have the whole index in that 2.5 Mb .gz file your plugin now returns results. Thank you!

Cheers

**Comment by harmtemolder** [comment]
1. Sorry, forgot to attach the actual release ZIP. Fixed that.
1. Mine are also ~2.5 and ~15 MB zipped and unzipped. I’ve changed it so that the index now contains new lines. It might become a bit bigger, but that does allow you to open it in a text editor, if you feel like it. Released that as [v0.2.2-alpha](https://github.com/harmtemolder/calibre-store-the-eye/releases/tag/v0.2.2-alpha), but search functionality hasn’t changed so no need to upgrade unless you really want to open the JSON file.
1. Good point, I’ve added a section about searching in the `README.md`. And you’re right, because you’re just searching  in a long list of URLs, there is no such thing as “Author” or “Keyword”.

Great to hear that everything works now!
